server/src/main/java/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -1156,8 +1156,24 @@ public ResourceCountCheckTask() {
         @Override
         protected void runInContext() {
             s_logger.info("Started resource counters recalculation periodic task.");
-            List<DomainVO> domains = _domainDao.findImmediateChildrenForParent(Domain.ROOT_DOMAIN);
-            List<AccountVO> accounts = _accountDao.findActiveAccountsForDomain(Domain.ROOT_DOMAIN);
+            List<DomainVO> domains;
+            List<AccountVO> accounts;
+            // try/catch task, otherwise it won't be rescheduled in case of exception
+            try {
+                domains = _domainDao.findImmediateChildrenForParent(Domain.ROOT_DOMAIN);
+            } catch (Exception e) {
+                s_logger.warn("Resource counters recalculation periodic task failed, unable to fetch immediate children for the domain " + Domain.ROOT_DOMAIN, e);
+                // initialize domains as empty list to do best effort recalculation
+                domains = new ArrayList<>();
+            }
+            // try/catch task, otherwise it won't be rescheduled in case of exception
+            try {
+                accounts = _accountDao.findActiveAccountsForDomain(Domain.ROOT_DOMAIN);
+            } catch (Exception e) {
+                s_logger.warn("Resource counters recalculation periodic task failed, unable to fetch active accounts for domain " + Domain.ROOT_DOMAIN, e);
+                // initialize accounts as empty list to do best effort recalculation
+                accounts = new ArrayList<>();
+            }
```

Comment:
does it make sense to keep running if one of these retrievals fail (for this pass I mean)?
I like the logging but i think a little further restructuring makes sense; two separate methods, one for domains and one for accounts for instance.
the recursion at lines 1197 and 1204 looks quite strange, if at all it works I would suggest a renaming for readability's sake.

Reply:
It is Shadowing in Java nested classes: https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html#shadowing

Reply:
yeah, well this doesn't make the code very readable. If there isn't a very strong design argument to do it this way, please get rid of it.

Reply:
```
does it make sense to keep running if one of these retrievals fail (for this pass I mean)?
```
Probably, I guess? At least we get the error logged now and if there is a problem with that specific account or domain's data it won't effect processing of the others.

Reply:
> does it make sense to keep running if one of these retrievals fail (for this pass I mean)?

Probably, I guess? At least we get the error logged now and if there is a problem with that specific account or domain's data it won't affect processing of the others.

> I like the logging but i think a little further restructuring makes sense; two separate methods, one for domains and one for accounts for instance.
> the recursion at lines 1197 and 1204 looks quite strange, if at all it works I would suggest a renaming for readability's sake.

I think changing the method names to be distinct has been addressed now, can we resolve the conversation?